two_points_on_the_same_branch.sh

```diff
@@ -119,6 +182,56 @@ if [ $(echo "$coq_opam_packages_on_separate_lines" | wc -l) != $(echo "$coq_opam
     exit 1
 fi
 
+# --------------------------------------------------------------------------------
+
+# Tell coqbot to update the initial comment, if we know which one to update
+function coqbot_update_comment() {
+    is_done="$1"
+    comment_body="$2"
+    uninstallable_packages="$3"
+
+    if [ ! -z "${coq_pr_number}" ]; then
+        comment_text=""
+
+        if [ -z "${is_done}" ]; then
+            comment_text="in progress, "
+        else
+            comment_text=""
+        fi
+        comment_text="Benchmarking ${comment_text}log available [here](${BUILD_URL}/console), workspace available [here](${JENKINS_URL}/view/benchmarking/job/${JOB_NAME}/ws/${BUILD_ID})"
+
+        if [ ! -z "${comment_body}" ]; then
+            comment_text="${comment_text}${nl}"'```'"${nl}${comment_body}${nl}"'```'
+        fi
+
+        if [ ! -z "${uninstallable_packages}" ]; then
+            comment_text="${comment_text}${nl}The following packages failed to install: ${uninstallable_packages}"
+        fi
+
+        # if there's a comment id, we update the comment while we're
+        # in progress; otherwise, we wait until the end to post a new
+        # comment
+        if [ ! -z "${coq_pr_comment_id}" ]; then
+            # XXX TODO Is this the right way to get the data out there?  Do we need to encode it?
+            curl -X POST --data-urlencode "${comment_text}" "${coqbot_url_prefix}/${coq_pr_number}/update-comment/${coq_pr_comment_id}"
+        elif [ ! -z "${is_done}" ]; then
+            # XXX TODO Is this the right way to get the data out there?  Do we need to encode it?
+            curl -X POST --data-urlencode "${comment_text}" "${coqbot_url_prefix}/${coq_pr_number}"
+        fi
+        if [ ! -z "${is_done}" ]; then
+            # XXX TODO FIXME How should we communicate this to coqbot?
+            curl -X POST "${coqbot_url_prefix}/${coq_pr_number}/benchmarking-done"
+        fi
```

Comment:
@Zimmi48 How should we be indicating these things to coqbot?

Reply:
I'd prefer to keep using static endpoints and put any additional information in the body of the request.

Reply:
What should I put into the body to indicate this?

Reply:
I am not sure I understood correctly your question. But the URL should be something static, like `/benchmarking-done` and the body should contain the information you need to pass to @coqbot. This could be the PR number on the first line, then the content to copy into the PR on the following lines. Anything that can be parsed easily is fine. JSON would be OK as well but it's probably not a good idea to generate JSON without a specialized tool.

Reply:
@Zimmi48 I've updated the urls to be static (I've used `/update-comment`, `/new-comment`, and `/benchmarking-done` to indicate that the `needs: benchmarking` label should be removed (there's a question of how to handle cases where the bench dies in the middle; do we want to leave `needs: benchmarking` in those cases, or do we want it removed when the bench is queued?)  I'm currently having the text be url-encoded, so I think you can just split on `&` and then split each entry on the first `=` and then url-un-encode the contents.

Reply:
> I'm currently having the text be url-encoded, so I think you can just split on `&` and then split each entry on the first `=` and then url-un-encode the contents.

Hum, more work for me then.

Reply:
Feel free to suggest any other encoding that supports newlines, etc.  This was the most common one that I saw used on the internet.  If you just want the text raw, separated by newlines, I can also make it do that.

Reply:
Yep, I prefer to have the data in the either JSON format, or raw and newline separated.

Reply:
@Zimmi48 I've updated it to use
```bash
# Tell coqbot to update the in-progress comment
curl -X POST --data-binary="${coq_pr_number}${nl}${coq_pr_comment_id}${nl}${comment_text}" "${coqbot_url_prefix}/update-comment"
```
```bash
# Tell coqbot to post a new comment that we're done benchmarking
curl -X POST --data-binary="${coq_pr_number}${nl}${comment_text}" "${coqbot_url_prefix}/new-comment"
```
and
```bash
# Tell coqbot to remove the `needs: benchmarking` label
curl -X POST --data-binary "${coq_pr_number}" "${coqbot_url_prefix}/benchmarking-done"
```
Does this work?